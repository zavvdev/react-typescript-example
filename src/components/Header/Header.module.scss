@import "src/styles/theme";

.root {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid $cGrayLight;
  height: 80px;

  .logo {
    height: 40px;
  }
}