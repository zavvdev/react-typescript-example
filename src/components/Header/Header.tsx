import Namespace from 'src/i18n/namespace';
import { useTranslation } from 'react-i18next';
import logo from 'src/assets/images/Logo.svg';
import LanguageSwitch from 'src/components/Header/components/LanguageSwitch/LanguageSwitch';
import classes from 'src/components/Header/Header.module.scss';

const Header = () => {
  const { t } = useTranslation(Namespace.Common);

  return (
    <header className={classes.root}>
      <img src={logo} className={classes.logo} alt={t('appName')} />
      <LanguageSwitch />
    </header>
  );
};

export default Header;
