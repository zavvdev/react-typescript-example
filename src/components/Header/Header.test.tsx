import { screen } from '@testing-library/react';
import render from 'src/tests/utils/render';
import Header from 'src/components/Header/Header';

describe('<Header />', () => {
  test('should render Header component (snapshot)', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have a logo', () => {
    render(<Header />);
    expect(screen.getByAltText('appName')).toBeInTheDocument();
  });

  test('should have language switch', () => {
    render(<Header />);
    expect(screen.getByText('en')).toBeInTheDocument();
    expect(screen.getByText('ru')).toBeInTheDocument();
  });
});
