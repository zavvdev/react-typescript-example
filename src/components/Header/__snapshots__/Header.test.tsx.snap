// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header component (snapshot) 1`] = `
<DocumentFragment>
  <header
    class="root"
  >
    <img
      alt="appName"
      class="logo"
      src="https://test-file-stub"
    />
    <div
      class="root"
    >
      <div
        class="langWrap"
      >
        <button
          class="lang active"
          type="button"
        >
          en
        </button>
        <span>
          |
        </span>
      </div>
      <div
        class="langWrap"
      >
        <button
          class="lang"
          type="button"
        >
          ru
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
